waiter/src/waiter/core.clj

```diff
@@ -692,12 +694,12 @@
    :start-scheduler-syncer-fn (pc/fnk [[:settings [:health-check-config health-check-timeout-ms failed-check-threshold] git-version]
                                        [:state clock]
                                        service-id->service-description-fn*]
-                                (let [http-client (http-utils/http-client-factory
-                                                    {:conn-timeout health-check-timeout-ms
-                                                     :socket-timeout health-check-timeout-ms
-                                                     :user-agent (str "waiter-syncer/" (str/join (take 7 git-version)))})
+                                (let [http-clients (http-utils/prepare-http-clients
+                                                     {:conn-timeout health-check-timeout-ms
+                                                      :socket-timeout health-check-timeout-ms
+                                                      :user-agent (str "waiter-syncer/" (str/join (take 7 git-version)))})
                                       available? (fn scheduler-available? [service-instance health-check-path]
-                                                   (scheduler/available? http-client service-instance health-check-path))]
+                                                   (scheduler/available? http-clients service-instance health-check-path))]
```

Comment:
Health checks triggered by Waiter target the primary port (`PORT0`) and need to use HTTP/2 protocol for health check invocations triggered on the main web server. Health checks triggered by Marathon or Kubernetes will target `PORT1` and use HTTP/1 protocol.